spec/graphql-pundit/instrumenter_spec.rb

```diff
@@ -15,63 +17,97 @@ def initialize(_, value)
     @value = value
   end
 
-  def create?
-    @value.to_s == "pass"
+  def test?
+    @value.to_s == 'pass'
   end
 end
 
-RSpec.describe GraphQL::Pundit::Instrumenter do
-  let(:instrumenter) { GraphQL::Pundit::Instrumenter.new }
-  let(:instrumented_field) { instrumenter.instrument(nil, field) }
-  let(:fail) { Test.new(:fail) }
-  let(:pass) { Test.new(:pass) }
-  let(:result) { instrumented_field.resolve(subject, {}, {})}
-
-  context 'authorize' do
-    let(:field) do
-      GraphQL::Field.define(type: "String") do
-        name "TestField"
-        authorize :create
-        resolve ->(obj, args, ctx) { obj.to_s }
-      end
+RSpec.shared_examples 'an authorizing field' do |error|
+  context 'Authorized' do
+    subject { pass_test }
+    it 'returns the value' do
+      expect(result).to eq('pass')
     end
+  end
 
-    context 'Authorized' do
-      subject { pass }
-      it 'returns the value' do
-        expect(result).to eq("pass")
+  context 'Unauthorized' do
+    subject { fail_test }
+    if error
+      it 'raises an execution error' do
+        expect { result }.to raise_error(GraphQL::ExecutionError)
       end
-    end
-
-    context 'Unauthorized' do
-      subject { fail }
+    else
       it 'returns nil' do
         expect(result).to eq(nil)
       end
     end
   end
+end
 
-  context 'authorize!' do
-    let(:field) do
-      GraphQL::Field.define(type: "String") do
-        name "TestField"
-        authorize! :create
-        resolve ->(obj, args, ctx) { obj.to_s }
+RSpec.shared_examples 'field with authorization' do |error|
+  context 'with query' do
+    context 'with name' do
+      let(:field) do
+        GraphQL::Field.define(type: 'String') do
+          name :test
```

Comment:
Is the name intended? As far as I understand the structure, you could name the field differently because you explicitly set the name of the Policy-test here.